src/main/java/com/github/dockerjava/core/async/AsyncResultCallback.java

```diff
@@ -0,0 +1,104 @@
+package com.github.dockerjava.core.async;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.concurrent.CountDownLatch;
+
+import com.github.dockerjava.api.async.ResultCallback;
+import com.google.common.base.Throwables;
+
+/**
+ * Implementation of {@link ResultCallback} with the single result event expected.
+ */
+public class AsyncResultCallback<A_RES_T> implements ResultCallback<A_RES_T> {
```

Comment:
This special callback class should inherit `ResultCallbackTemplate` ideally and should be moved into `InvocationBuilder` internally.